src/UID2.Client/UID2Encryption.cs

```diff
@@ -39,23 +41,36 @@ internal static DecryptionResponse Decrypt(string token, KeyContainer keys, Date
             {
                 return DecryptV2(Convert.FromBase64String(token), keys, now, domainName, clientType);
             }
-
-            if (data[1] == (int)AdvertisingTokenVersion.V3)
+            else
             {
-                return DecryptV3(Convert.FromBase64String(token), keys, now, identityScope, 3, domainName, clientType);
-            }
+                // Assume that newer tokens have an identity scope and type prefix
+                if (!ValidIdentityScopeAndType(data[0]))
+                {
+                    return DecryptionResponse.MakeError(DecryptionStatus.InvalidPayload);
```

Comment:
Can we add tests to cover the new errors?
And tests that take known invalid strings as input, such as a raw UID, a hashed DII, an entire JSON response (believe we've seen all 3)

Reply:
There was a test that already truncated tokens but it was returning invalid payload due to a decryption exception

This test is now exercising the length check

Added a new test to exercise the corrupt prefix check